stackit/internal/services/iaas/securitygroup/resource.go

```diff
@@ -0,0 +1,708 @@
+		return
+	}
+	diags = resp.State.Set(ctx, model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	tflog.Info(ctx, "security group updated")
+}
+
+// Delete deletes the resource and removes the Terraform state on success.
+func (r *securityGroupResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from state
+	var model Model
+	diags := req.State.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	projectId := model.ProjectId.ValueString()
+	securityGroupId := model.SecurityGroupId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "security_group_id", securityGroupId)
+
+	// Delete existing security group
+	err := r.client.DeleteSecurityGroup(ctx, projectId, securityGroupId).Execute()
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error deleting security group", fmt.Sprintf("Calling API: %v", err))
+		return
+	}
+
+	tflog.Info(ctx, "security group deleted")
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: project_id,security_group_id
+func (r *securityGroupResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, core.Separator)
+
+	if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing security group",
+			fmt.Sprintf("Expected import identifier with format: [project_id],[security_group_id]  Got: %q", req.ID),
+		)
+		return
+	}
+
+	projectId := idParts[0]
+	securityGroupId := idParts[1]
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "security_group_id", securityGroupId)
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), projectId)...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("security_group_id"), securityGroupId)...)
+	tflog.Info(ctx, "security group state imported")
+}
+
+func mapFields(ctx context.Context, securityGroupResp *iaasalpha.SecurityGroup, model *Model) error {
+	if securityGroupResp == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+
+	var securityGroupId string
+	if model.SecurityGroupId.ValueString() != "" {
+		securityGroupId = model.SecurityGroupId.ValueString()
+	} else if securityGroupResp.Id != nil {
+		securityGroupId = *securityGroupResp.Id
+	} else {
+		return fmt.Errorf("security group id not present")
+	}
+
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		securityGroupId,
+	}
+	model.Id = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+
+	var labels basetypes.MapValue
```

Comment:
refactor this to match what is implemented in https://github.com/stackitcloud/terraform-provider-stackit/blob/8894383c2c4e4f0036d061f594c0c40200de0aff/stackit/internal/services/iaas/volume/resource.go#L522

Reply:
refactored the code block